rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/mail/TbMsgToEmailNodeTest.java

```diff
@@ -15,98 +15,151 @@
+    private TbMsgToEmailNode node;
+    private TbMsgToEmailNodeConfiguration config;
+
+    private TbContext ctxMock;
+
+    @BeforeEach
+    void setUp() throws TbNodeException {
+        ctxMock = mock(TbContext.class);
+        originator = new DeviceId(UUID.randomUUID());
+        config = new TbMsgToEmailNodeConfiguration().defaultConfiguration();
+        node = new TbMsgToEmailNode();
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+    }
+
+    @AfterEach
+    void tearDown() {
+        node.destroy();
+    }
 
-    @Mock
-    private TbContext ctx;
+    @Test
+    void givenDefaultConfig_whenVerify_thenOK() {
+        // GIVEN-WHEN-THEN
+        assertThat(config.getFromTemplate()).isEqualTo("info@testmail.org");
+        assertThat(config.getToTemplate()).isEqualTo("${userEmail}");
+        assertThat(config.getSubjectTemplate()).isEqualTo("Device ${deviceType} temperature high");
+        assertThat(config.getBodyTemplate()).isEqualTo("Device ${deviceName} has high temperature $[temperature]");
+    }
 
-    private EntityId originator = new DeviceId(Uuids.timeBased());
-    private TbMsgMetaData metaData = new TbMsgMetaData();
-    private String rawJson = "{\"name\": \"temp\", \"passed\": 5 , \"complex\": {\"val\":12, \"count\":100}}";
+    @ParameterizedTest
+    @MethodSource("MailBodyTypeTestConfig")
+    public void givenMailBodyTypeTestConfig_whenOnMsg_thenVerify(MailBodyTypeTestConfig testConfig) throws TbNodeException {
+        // GIVEN
+        String mailBodyType = testConfig.getMailBodyType();
+        config.setMailBodyType(mailBodyType);
+        if (DYNAMIC_MAIL_BODY_TYPE.equals(mailBodyType)) {
+            config.setIsHtmlTemplate("${html}");
+        }
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        var md = new TbMsgMetaData();
+        md.putValue("userEmail", EXPECTED_TO_EMAIL);
+        md.putValue("deviceType", EXPECTED_DEVICE_TYPE);
+        md.putValue("deviceName", EXPECTED_DEVICE_NAME);
+        if (testConfig.getIsHtmlTemplateMdValue() != null) {
+            md.putValue("html", testConfig.getIsHtmlTemplateMdValue());
+        }
 
-    private RuleChainId ruleChainId = new RuleChainId(Uuids.timeBased());
-    private RuleNodeId ruleNodeId = new RuleNodeId(Uuids.timeBased());
+        var msgDataStr = "{\"temperature\": " + EXPECTED_TEMPERATURE + "}";
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, originator, md, msgDataStr);
+
+        // WHEN
+        node.onMsg(ctxMock, msg);
+
+        // THEN
+        var msgCaptor = ArgumentCaptor.forClass(TbMsg.class);
+        var typeCaptor = ArgumentCaptor.forClass(TbMsgType.class);
+        var originatorCaptor = ArgumentCaptor.forClass(EntityId.class);
+        var metadataCaptor = ArgumentCaptor.forClass(TbMsgMetaData.class);
+        var dataCaptor = ArgumentCaptor.forClass(String.class);
+        verify(ctxMock, times(1)).transformMsg(
```

Comment:
```suggestion
        verify(ctxMock).transformMsg(
```
![image](https://github.com/thingsboard/thingsboard/assets/99619831/966fb9ef-bafc-410d-b048-788991f14276)

verify is by default times 1